File: run_mandel2.js

console.log("Demo time!");

function bsp_putc(i) {
    var c = String.fromCharCode(i);
    process.stdout.write(c);
}

function put_float(v) {
    process.stdout.write(v)
}

const fs = require('fs');
var wasm_data = fs.readFileSync('./mandel2.wasm');

var module_ = new WebAssembly.Module(wasm_data);
var module = new WebAssembly.Instance(module_, {x: {putc: bsp_putc, put_float: put_float}});
module.exports.mandel();
// module.exports.mandel2();

File: run_mandel2.py

from ppci.wasm import instantiate, read_wasm

import mandel
import x
import time

with open('mandel2.wasm', 'rb') as f:
    module = read_wasm(f)

inst = instantiate(
    module,
    {
    'x': {
        'putc': x.putc,
        'put_float': x.put_float
    }
})

print(inst)
print('mandel():')
inst.exports['mandel']()
print()

print('mandel2():')
inst.exports['mandel2']()
print()
